core/util.go

```diff
@@ -317,24 +318,19 @@ func UniqueLowerNames(names []string) (unique []string) {
 	return
 }
 
-// UniqueLowerACMEIdentifiers returns the set of all unique ACME identifiers in
-// the input after all of them are lowercased. The returned identifier values
-// will be in their lowercased form.
-func UniqueLowerACMEIdentifiers(identifiers []identifier.ACMEIdentifier) (unique []identifier.ACMEIdentifier) {
-	identifierMap := make(map[identifier.ACMEIdentifier]struct{}, len(identifiers))
-	for _, id := range identifiers {
-		lowercasedID := identifier.ACMEIdentifier{
-			Type:  id.Type,
-			Value: strings.ToLower(id.Value),
-		}
-		identifierMap[lowercasedID] = struct{}{}
+// NormalizeIdentifiers returns the set of all unique ACME identifiers in the
+// input after all of them are lowercased. The returned identifier values will
+// be in their lowercased form and sorted alphabetically by value.
+func NormalizeIdentifiers(identifiers []identifier.ACMEIdentifier) []identifier.ACMEIdentifier {
+	for i := range identifiers {
+		identifiers[i].Value = strings.ToLower(identifiers[i].Value)
 	}
 
-	unique = make([]identifier.ACMEIdentifier, 0, len(identifierMap))
-	for id := range identifierMap {
-		unique = append(unique, id)
-	}
-	return
+	sort.Slice(identifiers, func(i, j int) bool {
+		return identifiers[i].Value < identifiers[j].Value
```

Comment:
I think the type should be part of the sort key
```suggestion
		return fmt.Sprintf("%s:%s", identifiers[i].Type, identifiers[i].Value) < fmt.Sprintf("%s:%s", identifiers[j].Type, identifiers[j].Value)
```